lib/Sema/CSDiagnostics.cpp

```diff
@@ -4823,13 +4823,27 @@ bool OutOfOrderArgumentFailure::diagnoseAsError() {
     auto text = SM.extractText(
         Lexer::getCharSourceRangeFromSourceRange(SM, firstRange));
 
-    auto removalRange =
-        SourceRange(Lexer::getLocForEndOfToken(
-                        SM, tuple->getElement(ArgIdx - 1)->getEndLoc()),
-                    firstRange.End);
+    SourceLoc removalStartLoc;
+    // For the first argumet it's always end of `(`.
+    if (ArgIdx == 0) {
+      removalStartLoc = tuple->getLParenLoc();
+    } else {
+      // For all argument arguments, start is next token past the previous
```

Comment:
Nit: I'm assuming "argument arguments" is a typo

Reply:
Heh, thank you!

Reply:
Fixed!